test/jdk/java/awt/Window/WindowIconUpdateOnDPIChanging/WindowIconUpdateOnDPIChangingTest.java

```diff
@@ -0,0 +1,147 @@
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JTextPane;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+import javax.swing.border.EmptyBorder;
+
+/* @test
+ * @bug 8276849
+ * @summary Update the window icon on DPI scale factor changes
+ * @run main/manual WindowIconUpdateOnDPIChangingTest
+ */
+public class WindowIconUpdateOnDPIChangingTest {
+
+    private static volatile boolean testResult = false;
+    private static final CountDownLatch countDownLatch = new CountDownLatch(1);
+    private static JFrame frame;
+
+    private static final String INSTRUCTIONS = "<html><body style=\"font-family: sans-serif\">\n"
+            + "<b>INSTRUCTIONS:</b>\n"
+            + "<p>Verify that the window icon is properly updated after changing the display DPI.</p>\n"
+            + "\n"
+            + "<p>The test is applicable for OSes that allows to change the display DPI\n"
+            + "without rebooting the system. Press <b>PASS</b> for other systems.</p>\n"
+            + "\n"
+            + "<ol>"
+            + "  <li>Set the display DPI scale factor to 100%</li>"
+            + "  <li>Check that the string \"16\" is painted in the window icon.</li>"
+            + "  <li>Set the display DPI scale factor to a different value,"
+            + "   or move the frame to another screen with a different DPI scale factor.</li>"
+            + "  <li>Check that the window icon is updated, it should display \"24\" on a screen"
+            + "      with a 150% DPI scaling, or \"32\" on a screen with a 200% DPI scaling."
+            + "      <br>If so, press <b>PASS</b>, otherwise press <b>FAIL</b>.</li>"
+            + "<ol>\n"
+            + "</body>\n"
+            + "</html>\n";
+
+    public static void main(String[] args) throws Exception {
+        SwingUtilities.invokeLater(WindowIconUpdateOnDPIChangingTest::createUI);
+        if (!countDownLatch.await(15, TimeUnit.MINUTES)) {
+            frame.dispose();
+            throw new RuntimeException("Timed out!");
+        } else if (!testResult) {
+            throw new RuntimeException("Test fails!");
+        }
+    }
+
+    private static void createUI() {
+        frame = new JFrame("Window Icon Update Test");
+        frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);
+        frame.setSize(640, 480);
+        frame.addWindowListener(new WindowAdapter() {
+            public void windowClosed(WindowEvent e) {
+                countDownLatch.countDown();
+            }
+        });
+        frame.getContentPane().setLayout(new BorderLayout());
+        frame.getContentPane().add(createInstrumentsPane(), BorderLayout.CENTER);
+        frame.getContentPane().add(createControlPanel(), BorderLayout.SOUTH);
+        frame.setIconImages(IntStream.rangeClosed(16, 32).mapToObj(size -> createIcon(size)).toList());
```

Comment:
In fact, I liked the version with method reference better even though the line is longer.

But I still suggest wrapping the line at each dot to make it clearer what operations are performed on the stream.
```suggestion
        frame.setIconImages(IntStream.rangeClosed(16, 32)
                                     .mapToObj(size -> createIcon(size))
                                     .toList());
```
It fits into 80 column limit. If it's replaced with method reference as it was before, it doesn't fit but I don't see it as big problem.

Alternatively, you can wrap the start of the stream too:
```suggestion
        frame.setIconImages(
                IntStream.rangeClosed(16, 32)
                         .mapToObj(WindowIconUpdateOnDPIChangingTest::createIcon)
                         .toList());
```
The line with method reference takes 82 columns, I'm sure it's acceptable.

Reply:
Looks like excessive wrapping to me for a very simple and quite readable one-liner.

Reply:
I don't think so: It makes the code easier to parse. Such formatting is very common with streams because each operation on the streams stands out, the person reading the code doesn't have to parse the long line in their mind.

I'm fine with it if everyone else agrees the code is too simple to require wrapping.